casper/src/test/scala/coop/rchain/casper/api/CreateBlockAPITest.scala

```diff
@@ -62,208 +57,209 @@ class CreateBlockAPITest extends FlatSpec with Matchers with EitherValues {
+  //          val engine                        = new EngineWithCasper[Task](n1.casperEff)
+  //          implicit val blockStore           = n1.blockStore
+  //          implicit val lastFinalizedStorage = n1.lastFinalizedStorage
+  //          implicit val synchronyConstraintChecker =
+  //            SynchronyConstraintChecker[Effect](syncConstraintThreshold)
+  //          implicit val lastFinalizedHeightConstraintChecker =
+  //            LastFinalizedHeightConstraintChecker[Effect](Long.MaxValue)
+  //          for {
+  //            deploys      <- (0 until 3).toList.traverse(i => basicDeployData[Task](i))
+  //            engineCell   <- Cell.mvarCell[Task, Engine[Task]](engine)
+  //            blockApiLock <- Semaphore[Effect](1)
+  //
+  //            b1 <- n1.publishBlock(deploys(0))(nodes: _*)
+  //            b2 <- n2.publishBlock(deploys(1))(nodes: _*)
+  //            _  <- n1.syncWith(n2)
+  //
+  //            _        <- n1.casperEff.addDeploy(deploys(2))
+  //            b3Status <- createBlock(blockApiLock)(engineCell)
+  //
+  //            _ = b3Status.left.value should include(
+  //              "Must wait for more blocks from other validators"
+  //            )
+  //          } yield ()
+  //      }
+  //  }
+  //
+  //  it should "allow proposals with enough new blocks from other validators" in effectTest {
+  //    TestNode
+  //      .networkEff(genesis, networkSize = 5, synchronyConstraintThreshold = syncConstraintThreshold)
+  //      .use {
+  //        case nodes @ n1 +: n2 +: n3 +: _ +: _ +: Seq() =>
+  //          import n1.{logEff, metricEff, span, timeEff}
+  //          val engine                        = new EngineWithCasper[Task](n1.casperEff)
+  //          implicit val blockStore           = n1.blockStore
+  //          implicit val lastFinalizedStorage = n1.lastFinalizedStorage
+  //          implicit val synchronyConstraintChecker =
+  //            SynchronyConstraintChecker[Effect](syncConstraintThreshold)
+  //          implicit val lastFinalizedHeightConstraintChecker =
+  //            LastFinalizedHeightConstraintChecker[Effect](Long.MaxValue)
+  //          for {
+  //            deploys      <- (0 until 4).toList.traverse(i => basicDeployData[Task](i))
+  //            engineCell   <- Cell.mvarCell[Task, Engine[Task]](engine)
+  //            blockApiLock <- Semaphore[Effect](1)
+  //
+  //            b1 <- n1.publishBlock(deploys(0))(nodes: _*)
+  //            b2 <- n2.publishBlock(deploys(1))(nodes: _*)
+  //            b3 <- n3.publishBlock(deploys(2))(nodes: _*)
+  //            _  <- n1.syncWith(n2, n3)
+  //
+  //            _        <- n1.casperEff.addDeploy(deploys(3))
+  //            b4Status <- createBlock(blockApiLock)(engineCell)
+  //
+  //            _ = b4Status shouldBe a[Right[_, String]]
+  //          } yield ()
+  //      }
+  //  }
+  //
+  //  it should "check for new deploys before checking synchrony constraint" ignore effectTest {
```

Comment:
With the Proposer we still need to check synchrony constraint, is this done in some other test?

Reply:
https://github.com/rchain/rchain/issues/3277#issuecomment-798109618